flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java

```diff
@@ -84,6 +85,9 @@ public class TemporalRowTimeJoinOperator extends BaseTwoInputStreamOperatorWithS
     private static final String RIGHT_STATE_NAME = "right";
     private static final String REGISTERED_TIMER_STATE_NAME = "timer";
     private static final String TIMERS_STATE_NAME = "timers";
+    private static final String RIGHT_LATE_ELEMENTS_METRIC_NAME = "rightNumLateRecords";
+
+    private transient Counter rightNumLateRecords;
```

Comment:
I notice that we already have numLateRecordsDropped metric. Is this sufficient? 
You are proposing a new metric for a subset of operators (ones with 2 inputs) and only for one side, where all the processing is the same for the left and right sides at the moment.  
I wonder if the metric should be when the late record is dropped in line with the existing metric
We would need documentation to be added for this.